File: tauri-agent-demo/src/shared/api/internal/teams.ts
import type { Team, TeamHandoffEvent } from '../../../types';
import { API_BASE_URL } from './base';

export async function getTeam(teamId: string): Promise<Team> {
    const response = await fetch(`${API_BASE_URL}/teams/${teamId}`);
    if (!response.ok) throw new Error('Failed to fetch team');
    return response.json();
}

export async function getTeamHandoffs(teamId: string): Promise<TeamHandoffEvent[]> {
    const response = await fetch(`${API_BASE_URL}/teams/${teamId}/handoffs`);
    if (!response.ok) throw new Error('Failed to fetch team handoffs');
    return response.json();
}

File: tauri-agent-demo/python-backend/server/routers/teams.py
from fastapi import APIRouter

from models import Team, TeamHandoffEvent
from server.services import team_service


router = APIRouter(tags=["teams"])
router.add_api_route("/teams/{team_id}", team_service.get_team, methods=["GET"], response_model=Team)
router.add_api_route("/teams/{team_id}/handoffs", team_service.get_team_handoffs, methods=["GET"], response_model=list[TeamHandoffEvent])
